### PR TITLE
expand on to_string() documentation

### PR DESCRIPTION
**Pull request by polettix**
As anticipated on #792, a (minimal(istic)) proposal for a better documentation of to_string()

### CODE REVIEW

File: lib/Mojo/Message.pm [hunk 1]
```diff
@@ -630,7 +630,10 @@ L</"default_charset">.
 
   my $str = $msg->to_string;
 
-Render whole message.
+Render whole message. For performance optimization reasons, calling this
+method freezes the message representation, so that subsequent calls
+to other methods (e.g. to modify the headers) will have no effect on the
+message rendering.
```
Comment (kraih) [inline lib/Mojo/Message.pm]: This is incorrect, only certain parts of the message might get frozen, and we make no guarantees which. (It's an optimization after all, and we need to be able to keep changing it!)

**Comment by kraih** [comment]
:-1: from me, because the description is incorrect.

**Comment by tempire** [comment]
It might be a better idea to give a warning that to_string might freeze parts of the message, instead of documenting exactly what it does, so that the dev is prompted to do further research if they need it, or find another method to do what they want.

It would be a shame if optimizations change and the docs are forgotten or out of date for a period of time.

**Comment by kraih** [comment]
And don't forget that it might not even be possible to render a message twice, we really should avoid making it look like using `to_string` is a good idea.

**Pull request closed without merging**